masonry/src/render_root.rs

```diff
@@ -501,44 +453,28 @@ impl RenderRoot {
         kurbo::Size::new(size.width, size.height)
     }
 
-    // --- MARK: POST-EVENT ---
-    fn post_event_processing(&mut self, widget_state: &mut WidgetState) {
-        // If children are changed during the handling of an event,
-        // we need to send RouteWidgetAdded now, so that they are ready for update/layout.
-        if widget_state.children_changed {
-            run_update_new_widgets_pass(self);
-        }
-
-        if self.state.debug_logger.layout_tree.root.is_none() {
-            self.state.debug_logger.layout_tree.root = Some(self.root.id().to_raw() as u32);
-        }
-
-        if !self.state.scroll_request_targets.is_empty() {
-            run_update_scroll_pass(self);
-        }
-
-        if self.root_state().needs_compose && !self.root_state().needs_layout {
-            root_compose(self, widget_state);
-        }
-
-        // Update the disabled and stashed state if necessary
-        // Always do this before updating the focus-chain
-        if self.root_state().needs_update_disabled {
```

Comment:
I think these should also be either in the pass function below this line, or in `run_rewrite_passes`? (same for the other  `self.root_state().needs_...` below)

Reply:
Each pass handles its own early-exit. I think that's cleaner than having a lot of sort-circuit code in run_rewrite_passes, for a few reasons:

- It makes the run_rewrite_passes function shorter and very easy to read.
- In a lot of cases the early exit is part of the recursive algorithm, so doing an early-exit in run_rewrite_passes ends up duplicating code.
- I think a general rule of "all passes do their own early exit" is easier to remember than "these passes do their own early exit, these passes don't and need to be called within a short-circuit block".

I should probably document this in the code though.

Reply:
Ok, I guess I missed that.